### PR TITLE
Xamarin Support & Component

### PR DESCRIPTION
**Pull request by mattleibow**
I added the Xamarin.Android and Xamairn.iOS to the nuspec as this library works there as well. No actual code has changed to do this. I did, however, have to create a new project file for Xamarin.iOS and only include the .NET 2.0 PCL code. Currently, Xamarin.iOS does not support Expression compilation and throws a runtime exception.

Are you interested in adding an item to the Xamarin Component Store? The component is actually just a reference to the NuGet and includes samples and some basic docs. I have added all the required bits (the samples, the icons, the docs and the component definition file) and all that needs to be done is run the .bat file.

To submit, you just have to create a free Xamarin account (https://xamarin.com/account/register) and submit to the store (https://components.xamarin.com/submit). The component will be reviewed and then published by one of the guys there.

Submitting to the store allows greater visibility for mobile developers :)

Xamarin Component Docs:
- [Xamarin Component Submission](http://developer.xamarin.com/guides/cross-platform/advanced/submitting_components/component_submission_quickstart/)
- [Detailed Guide](http://developer.xamarin.com/guides/cross-platform/advanced/submitting_components/component_submission_guide/)

### CODE REVIEW

File: src/CsvHelper.nuspec [hunk 1]
```diff
@@ -18,6 +18,7 @@
 		<file src="CsvHelper\bin\Release\CsvHelper.*" target="lib\net40-client\" />
 		<file src="CsvHelper20\bin\Release\CsvHelper.*" target="lib\net20\" />
 		<file src="CsvHelper35\bin\Release\CsvHelper.*" target="lib\net35-client\" />
-		<file src="CsvHelperPcl\bin\Release\CsvHelper.*" target="lib\portable-sl4+wp71+windows8+net40\" />
+		<file src="CsvHelperPcl\bin\Release\CsvHelper.*" target="lib\portable-sl4+wp71+windows8+net40+wpa81+monoandroid\" />
+		<file src="CsvHelperXamarinPcl\bin\Release\CsvHelper.*" target="lib\portable-sl4+wp71+windows8+net40+wpa81+monoandroid+monotouch+xamarinios\" />
```
Comment (mattleibow) [inline src/CsvHelper.nuspec]: Just added the Xamarin platforms to the supported list

File: src/CsvHelper.nuspec [hunk 2]
```diff
@@ -18,6 +18,7 @@
 		<file src="CsvHelper\bin\Release\CsvHelper.*" target="lib\net40-client\" />
 		<file src="CsvHelper20\bin\Release\CsvHelper.*" target="lib\net20\" />
 		<file src="CsvHelper35\bin\Release\CsvHelper.*" target="lib\net35-client\" />
-		<file src="CsvHelperPcl\bin\Release\CsvHelper.*" target="lib\portable-sl4+wp71+windows8+net40\" />
+		<file src="CsvHelperPcl\bin\Release\CsvHelper.*" target="lib\portable-sl4+wp71+windows8+net40+wpa81+monoandroid\" />
```
Comment (mattleibow) [inline src/CsvHelper.nuspec]: This one would have worked, but the use of Expressions fail on iOS

File: src/CsvHelper/CsvParser.cs [hunk 3]
```diff
@@ -117,7 +116,7 @@ public virtual string[] Read()
 				if( configuration.DetectColumnCountChanges && row != null )
 				{
 					if( FieldCount > 0 && ( FieldCount != row.Length || 
-#if NET_2_0
+#if NET_2_0 && !PCL
```
Comment (mattleibow) [inline src/CsvHelper/CsvParser.cs]: There are a few of these to include support for PCL, but using the .NET 2.0 profile - used for iOS only

File: src/CsvHelperXamarinPcl/CsvHelperXamarinPcl.csproj [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    <MinimumVisualStudioVersion>10.0</MinimumVisualStudioVersion>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{865C2EAE-7A31-4015-ADD8-DD32027342CF}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>CsvHelper</RootNamespace>
+    <AssemblyName>CsvHelper</AssemblyName>
+    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFrameworkProfile>Profile136</TargetFrameworkProfile>
```
Comment (mattleibow) [inline src/CsvHelperXamarinPcl/CsvHelperXamarinPcl.csproj]: This project is exactly the same as the other portable one, but only has the files as the .NET 2.0 project

File: src/component.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+%YAML 1.2
+---
+
+# package info
+id: csv-helper
+version: 2.10.0
+name: CSV Helper
```
Comment (mattleibow) [inline src/component.yaml]: This file contains the Xamarin Component definition - similar to a nuspec for NuGet

File: src/component.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+#  - screenshots/screenshot1.png
+icons: 
+  - ../logo/csv-helper_128x128.png
+  - ../logo/csv-helper_512x512.png
+
+# samples
+samples: 
+  iOS Unified Sample: ../samples/iOS/CsvHelperSample-ios.sln
+  Android Sample: ../samples/Android/CsvHelperSample-android.sln
+  Windows Phone Sample: ../samples/WindowsPhone/CsvHelperSample-wp.sln
```
Comment (mattleibow) [inline src/component.yaml]: Samples are required, but can be very simple

File: src/component.yaml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+# docs
+license: ../docs/License.md
+getting-started: ../docs/GettingStarted.md
+details: ../docs/Details.md
```
Comment (mattleibow) [inline src/component.yaml]: Docs are required, but can be simple and point to other docs

File: src/component.yaml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+# package dependencies
+no_build: true
+is_shell: true
+packages:
+  mobile: CsvHelper, Version=2.10
```
Comment (mattleibow) [inline src/component.yaml]: This bit is the actual contents:

 - `no_build` tells the package not to build all the solutions in the directory (can be removed)
 - `is_shell` tells the IDE to just install the NuGet
 - `packages` lists the NuGets to install

**Comment by JoshClose** [comment]
This is A LOT to take in. :) I think having this in the Xamarin components store would be useful for people. It may be a bit before I can do this though. I want to understand everything that is going on before I put it in. I will have to get my environment setup, and find some time too.

**Comment by mattleibow** [comment]
No problem.
Let me know if you have any problems, questions or suggestions about anything!
You can contact me via email on matthew.leibowitz@xamarin.com or mattleibow@live.com. Or, on this issue; I'll be watching.

**Comment by JoshClose** [comment]
Thanks Matt!

**Comment by JoshClose** [comment]
I pulled this down locally and had to fix some conflicts due to changes that occurred after. I pushed the branch up. Can you take a look at make sure everything looks good to you? If so, I'll merge it in.

I'm guessing I should wait on this until I pull the other pull request.

**Comment by JoshClose** [comment]
Has this story changed since MS took over?

**Comment by danmiser** [comment]
Yes, things have changed quite a bit with the Xamarin Component Store. It is no longer running and they recommend upgrading from Component Store to nuget. So this one should be closed out.
https://docs.microsoft.com/en-us/xamarin/cross-platform/troubleshooting/component-nuget?tabs=macos

**Comment by JoshClose** [comment]
Thanks.

**Pull request closed without merging**